jobwatcher/jobwatcher.py

```diff
@@ -51,17 +52,56 @@ def get_asg_name(stack_name, region, proxy_config):
 
     return asg_name
 
-def get_instance_properties(instance_type):
+
+def read_cfnconfig():
+    cfnconfig_params = {}
+    with open(cfnconfig_file) as f:
+        for kvp in f:
+            key, value = kvp.partition('=')[::2]
+            cfnconfig_params[key.strip()] = value.strip()
+    return cfnconfig_params
+
+
+def get_vcpus_from_pricing_file(instance_type):
     with open(pricing_file) as f:
         instances = json.load(f)
         try:
-            slots = int(instances[instance_type]["vcpus"])
-            log.info("Instance %s has %s slots." % (instance_type, slots))
-            return {'slots': slots}
+            vcpus = int(instances[instance_type]["vcpus"])
+            log.info("Instance %s has %s vcpus." % (instance_type, vcpus))
+            return vcpus
         except KeyError as e:
             log.error("Instance %s not found in file %s." % (instance_type, pricing_file))
             exit(1)
 
+
+def get_instance_properties(instance_type):
+    cfnconfig_params = read_cfnconfig()
+    try:
+        cfn_scheduler_slots = cfnconfig_params["cfn_scheduler_slots"]
+        slots = 0
+        if cfn_scheduler_slots.isdigit():
+            slots = int(cfn_scheduler_slots)
+            log.info("Instance %s will use %s slots based on configuration." % (instance_type, slots))
+        else:
+            vcpus = get_vcpus_from_pricing_file(instance_type)
+            if cfn_scheduler_slots is "cores":
+                log.info("Instance %s will use number of cores as slots based on configuration." % instance_type)
+                slots = vcpus / 2
```

Comment:
what if there's 1 vcpu?

Reply:
I could not find an instance with 1 vcpu? How does the cookbook handle that?

Reply:
There's a couple single vcpu t2's. You should set the slots as `1` in that case.

https://aws.amazon.com/ec2/instance-types/

Reply:
I am still interested in knowing what the cookbook does in that case. Maybe we are carrying a bug in the cookbook. But, I will make that change.

Reply:
Done. This will be done in the next PR. Fixed the cookbook bug separately in https://github.com/awslabs/cfncluster-cookbook/pull/171